tasks/tangd-custom-port.yml

```diff
@@ -0,0 +1,53 @@
+---
+# This task tells SELinux for the port that we want the tangd service to use
+- name: Allow the custom port for tangd_port_t in SELinux
+  community.general.seport:
+    ports: "{{ nbde_server_port }}"
+    proto: tcp
+    setype: tangd_port_t
+    state: present
+  when: ansible_selinux.status == "enabled"
+
+# This block creates the override file for systemd with the new port that we have requested
+- name: Create override file
+  block:
+    # This task checks if the directory /etc/systemd/system/tangd.socket.d exist and registers the value in systemd__system__tangd_socket
+    - name: Check if directory /etc/systemd/system/tangd.socket.d exist
+      stat:
+        path: /etc/systemd/system/tangd.socket.d
+      register: systemd_system_tangd_socket
+
+    # This tasks Create the /etc/systemd/system/tangd.socket.d directory if it does not exist
+    - name: Create a directory if it does not exist
+      file:
+        path: /etc/systemd/system/tangd.socket.d
+        state: directory
+        mode: '0755'
+      when: systemd_system_tangd_socket.stat.exists
+
+    # This tasks creates the file with the port entry that we want tangd to listen to 
+    - name: Creates the file with the port entry that we want tangd to listen to 
+      blockinfile:
+        path: /etc/systemd/system/tangd.socket.d/override.conf
+        block: |
+          [Socket]
+          ListenStream={{ nbde_server_port }}
+        state: present
+        create: yes
+        mode: '0644'
+        marker: "# {mark} ANSIBLE MANAGED BLOCK NBDE"
+    
+    # This tasks reload the daemons so the new changes take effect
+    - name: Reload the daemons so the new changes take effect
+      systemd:
+        daemon_reload: yes
```

Comment:
This should be conditional i.e. `when: __nbde_server_daemon_reload` - then one of the tasks above should set `__nbde_server_daemon_reload` if the daemon needs to be reloaded